slides/linear_model.md

```diff
@@ -0,0 +1,260 @@
+
+
+
+---
+# Linear regression
+
+The linear regression objectif is to minimize the distance between the prediction curve and the datapoints
+
+.shift-down.pull-left.shift-left[<img src="../figures/linear_fit_red.svg" width="100%">]
+
+???
+An error for each point correspond to the red bar in the figure
+the best fit is the line which minimize the sum of (the square of) those red lines.
+
+Fortunatly, there is a formula, given **X** and **y**, to find the optimal weights in an efficient manner
+
+---
+# Linear regression in higher dimension
+
+.pull-left.shift-left[<img src="../figures/lin_reg_3D.svg" width="130%">]
+
+
+???
+In real world data set, **X** has severals dimensions, and it is not possible anymore to represent it.
+
+---
+# For classification: logistic regression
+
+For **classification** we use a logistic regression model  
+**y** is binary, either +1 or -1
+
+.shift-left.pull-left[<img src="../figures/categorical.svg" width="100%">]
+.pull-right.very-small[
+```python
+From sklearn.linear_model import LogisticRegression
+log_reg = LogisticRegression()
+```
+]
+???
+Logistic regression is a linear model for **classification** - not regresion as the name suggest.
+
+In our adult_census dataset, we do not have continous value for salary but only whether the salary is higher than $50K or not.
+
+
+---
+# For classification: logistic regression
+
+The output is now modeled as a form of step function, which is adjusted on the data.
+
+.shift-left.pull-left[<img src="../figures/logistic_color.svg" width="100%">]
+.pull-right.very-small[
+```python
+From sklearn.linear_model import LogisticRegression
+log_reg = LogisticRegression()
+log_reg.fit(X, y)
+```]
+
+---
+# Logistic regression in 2 dimensions
+
+**X** is 2-dimensional  
+**y** is the color
+
+.shift-left.pull-left[<img src="../figures/logistic_2D.svg" width="100%">]
+.pull-right[<img src="../figures/logistic_3D.svg" width="100%">]
+
+???
+Here is an other way of representing our data.
+In this case, X has two dimension x1 and x2.
+The axis correspond to x1, x2
+and the color correspond to the target label y 
+
+---
+# Model complexity
+Salary = *.4 x* education_level + *.2 x* hours-per-week + *.1 x* age + 
+.red[ *.2 x* zodiac_sign + *.1 x* red_socks] + ...  
+
+* A linear model can also overfit. 
+
+The most classic way to limit its complexity is to push coefficients toward small values. Such model is called *Ridge* for regression settings. 
+???
+
+If we have too many parameters w.r.t. number of samples, it's adviced to penalize the parameters of our models.
+
+With weights penalty, we include the value of the model's weights within the objectif function
+So a penalized model should choose lower weights for almost a similar fit
+
+The complexity parameter \alpha controls the amount of shrinkage: the larger the value of \alpha, the greater the amount of shrinkage and thus the coefficients become more robust to collinearity.
```

Comment:
```suggestion
In this penalized objective function, a complexity parameter allows to control
the amount of shrinkage and it is denominated \alpha. The larger the value of
\alpha, the greater the amount of shrinkage and thus the coefficients become
more robust to collinearity.
```

Reply:
2 questions here:

* is it only collinearity?
* we introduce `alpha` without equation nor code. I would think that having a `Ridge` model showing the `alpha` there could be good?

Reply:
My goal in this MOOC is to avoid equations as much as possible.